src/test/scala/scorex/core/network/peer/InMemoryPeerDatabaseSpec.scala

```diff
@@ -103,4 +105,51 @@ class InMemoryPeerDatabaseSpec extends NetworkTests {
     }
   }
 
+  it should "not add more peers than configured by storedPeersLimit" in {
+    withDb { db =>
+      val ninePeers = generatePeers(9)
+      val oneMorePeer = generatePeers(1)
+      val fiveMorePeers = generatePeers(5)
+      db.addOrUpdateKnownPeers(ninePeers)
+      db.blacklistedPeers.isEmpty shouldBe true
+      db.knownPeers.size shouldBe 9
+
+      db.addOrUpdateKnownPeers(oneMorePeer)
+      db.blacklistedPeers.isEmpty shouldBe true
+      db.knownPeers.size shouldBe 10
+
+      db.addOrUpdateKnownPeers(fiveMorePeers)
+      db.blacklistedPeers.isEmpty shouldBe true
+      db.knownPeers.size shouldBe 10
+    }
+  }
+
+  it should "not replace existing peers if they are connected" in {
```

Comment:
let's add the test that in verify the replacement of the "oldest" non-connected peers.